// Copyright 2024 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

package secretflow.serving.tools;

message InferenceConfig {
  // Inference requester id, inference result file is only output in requester
  // party
  string requester_id = 1;

  // The file path of inference.
  string result_file_path = 2;

  // Optional. Additional columns must exist in the requester's feature input
  // file and are added to the output file.
  repeated string additional_col_names = 3;

  // The name of inference score column name in result file.
  string score_col_name = 4;

  // Optional. This determines the size of each request batch.
  int32 block_size = 11;
}
